botocore/client.py

```diff
@@ -617,6 +618,10 @@ class Config(object):
         thrown when attempting to read from a connection. The default is
         60 seconds.
 
+    :type parameter_validation: str
+    :param parameter_validation: Whether parameter validation should occur
+        when serializing requests. The default is True.
+
```

Comment:
My only concern for this PR is whether the caveats so kindly laid out by @JordonPhillips in #796 should be somehow mentioned here or not.